pkg/controller/pipelinegates/reconciler.go

```diff
@@ -148,47 +140,37 @@ func (s *GateReconciler) Reconcile(ctx context.Context, id string) (result recon
 		return reconcile.Result{}, err
 	}
 
-	if err = s.CheckNamespace(gateCR.Namespace); err != nil {
-		logger.Error(err, "failed to check namespace")
-		return reconcile.Result{}, err
-	}
-
 	// get pipelinegate
 	currentGate := &v1alpha1.PipelineGate{}
 	if err := s.K8sClient.Get(ctx, types.NamespacedName{Name: gateCR.Name, Namespace: gateCR.Namespace}, currentGate); err != nil {
-		if apierrors.IsNotFound(err) {
-			logger.V(1).Info("This gate doesn't yet have a corresponding CR on this cluster yet.", "Namespace", gateCR.Namespace, "Name", gateCR.Name, "ID", gateCR.Spec.ID)
-			// If the PipelineGate doesn't exist, create it.
-			err = s.K8sClient.Create(context.Background(), gateCR)
-			if err != nil {
-				logger.Error(err, "failed to create gate", "Namespace", gateCR.Namespace, "Name", gateCR.Name, "ID", gateCR.Spec.ID)
-				return reconcile.Result{}, err
-			}
-		}
-		logger.V(1).Info("Could not get gate.", "Namespace", gateCR.Namespace, "Name", gateCR.Name, "ID", gateCR.Spec.ID)
-	} else {
-		logger.V(1).Info("Gate exists.", "Namespace", gateCR.Namespace, "Name", gateCR.Name, "ID", gateCR.Spec.ID)
-		scope, _ := pgctrl.NewPipelineGateScope(ctx, s.K8sClient, currentGate)
-
-		// reset job ref to trigger a rerun
-		// do NOT reset state to pending, as this will happen in the reconciler to make sure we transition from open/closed to pending
-		if (currentGate.Status.IsClosed() || currentGate.Status.IsOpen()) && client.IsPending(gate) && !client.HasJobRef(gate) {
-			currentGate.Status.SetJobRef("", "")
-			currentGate.Spec = gateCR.Spec
+		if !apierrors.IsNotFound(err) {
+			logger.V(1).Info("Could not get gate.", "Namespace", gateCR.Namespace, "Name", gateCR.Name, "ID", gateCR.Spec.ID)
+			return reconcile.Result{}, err
 		}
 
-		err = scope.PatchObject()
-		if err != nil {
-			logger.Error(err, "failed to patch gate", "Namespace", gateCR.Namespace, "Name", gateCR.Name, "ID", gateCR.Spec.ID)
+		logger.V(1).Info("This gate doesn't yet have a corresponding CR on this cluster yet.", "Namespace", gateCR.Namespace, "Name", gateCR.Name, "ID", gateCR.Spec.ID)
+		// If the PipelineGate doesn't exist, create it.
+		if err = s.K8sClient.Create(context.Background(), gateCR); err != nil {
+			logger.Error(err, "failed to create gate", "Namespace", gateCR.Namespace, "Name", gateCR.Name, "ID", gateCR.Spec.ID)
 			return reconcile.Result{}, err
 		}
+		return reconcile.Result{}, nil
 	}
 
-	return
-}
+	logger.V(1).Info("Gate exists.", "Namespace", gateCR.Namespace, "Name", gateCR.Name, "ID", gateCR.Spec.ID)
+	scope, err := pgctrl.NewPipelineGateScope(ctx, s.K8sClient, currentGate)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	currentGate.Spec = gateCR.Spec
+	err = scope.PatchObject()
```

Comment:
Why do we still need to patch the object here?

Also I don't believe you should be modifying the console gate spec from the agent, just the status (eg the job it points to and it's status entry)

Reply:
yes, it's not needed. We can only create the CRD here

Reply:
done